### ISSUE TITLE
Pattern for extracting information from the context

### ISSUE THREAD

**Issue by bprosnitz** [vanadium/issues#796]
This came up during our API usability workshop, so I wanted to create an issue for it.

The user needs to know where to go to extract information from the context and how to use it. It is not evident from the object itself how it can be used and what functionality it offers.

**Comment by asimshankar** [comment]
Are you talking about *context.T?
I believe it is intentionally designed as an arbitrary map of keys and values where packages can provide functions to set and get the values. There is no fixed set.

Is it that the documentation at https://godoc.v.io/pkg/v.io/v23/context/ isn't clear about this?

Adding @mattr-google - the owner of the context :)

**Comment by bprosnitz** [comment]
Yes *context.T -- I'm just relaying comments that came up when discussing API usability papers.

The usability challenge is that it can be hard to find what you can do with the object and how. It seems that the most common cases are to use a small set of actions and users need to know where to find the appropriate methods and know to look for them to perform the actions. There may be a way to make them easier to discover and remember. For instance, if there were string keys you could dump the object and see what is there. Or, if there was some other way to access the methods more directly on it. 

We don't necessarily have to do anything about this, but it does seem that people will currently run into situations where it is unclear how to do particular things and not realize that they can extract it from the context so this issue is mostly to bring up this problem.

**Comment by asimshankar** [comment]
Does the godoc for this not cover it: https://godoc.v.io/pkg/v.io/v23/context/#pkg-index ?
I'm curious how this documentation factored into the usability study and what other ways there are to improve discoverability

**Comment by asimshankar** [comment]
Reassigning back to @bprosnitz to get his attention :)
